packages/components/src/spectrum/View.tsx

```diff
@@ -23,13 +30,46 @@ export type ViewProps = Omit<SpectrumViewProps<6>, 'backgroundColor'> & {
 
 export const View = forwardRef<DOMRefValue<HTMLElement>, ViewProps>(
   (props, forwardedRef): JSX.Element => {
-    const { backgroundColor, UNSAFE_style, ...rest } = props;
+    const {
+      backgroundColor,
+      borderColor,
+      borderStartColor,
+      borderEndColor,
+      borderTopColor,
+      borderBottomColor,
+      borderXColor,
+      borderYColor,
+      UNSAFE_style,
+      ...rest
+    } = props;
+
+    // Using shorthand to define border styling. Removing a style property during rerender can lead to styling bugs
+    // ex. changing from borderXColor = blue to borderColor = blue will cause the left and right borders to disappear
```

Comment:
Weird, I can't find the error you mentioned in the [React spectrum source code](https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/utils/src/styleProps.ts) at all. But the error also says "replace the shorthand with separate values", which is the opposite of what you're doing here (you're converting everything to [shorthand](https://developer.mozilla.org/en-US/docs/Web/CSS/Shorthand_properties)).
Also, we shouldn't be setting these properties at all if they're not set; leave them `undefined`, not `'transparent'`.

Reply:
My bad on this one I think I was misusing the useMemo hook earlier. The new way I define borderStyle defines everything into separate values and unset properties are no longer set to transparent